command/agent/http.go

```diff
@@ -531,13 +531,7 @@ func (s *HTTPServer) parseToken(req *http.Request, token *string) {
 // DC in the request, if given, or else the agent's DC.
 func (s *HTTPServer) parseSource(req *http.Request, source *structs.QuerySource) {
 	s.parseDC(req, &source.Datacenter)
-	if node := req.URL.Query().Get("near"); node != "" {
```

Comment:
Still think you'll need to revert this change - should not be needed any more and there are other RPC endpoints that do the sort which rely on `_agent` getting resolved on the agent side, not the server side.